src/Symfony/Component/Security/Http/Firewall/ContextListener.php

```diff
@@ -103,6 +106,10 @@ public function handle(GetResponseEvent $event)
 
         if ($token instanceof TokenInterface) {
             $token = $this->refreshUser($token);
+
+            if (null === $token && null !== $this->rememberMeServices) {
+                $this->rememberMeServices->loginFail($request);
```

Comment:
In case of an anonymous token, the user will be `anon.`, causing this case to not get hit. If I understand this case correctly, it will simply "postpone" the logout until a higher authentication level is required.

I don't think this is a problem, right?

Reply:
I don't think it's a problem either.

Reply:
The `AnonymousToken` is not stored in the session (https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Security/Http/Firewall/ContextListener.php#L142), and its very unlikely it would be returned by `refreshUser`, so that really shouldn't be a problem 🙃